### ISSUE TITLE
Doesn't work.  

### ISSUE THREAD

**Issue by dellablue** [aortbals/southwest-checkin#68]
Doesn't work.  The application successfully found my flight when I queried by inputting locator id and name.  And the app said it would check me in.  But it didn't.  Nor did it work for a friend of mine a couple of weeks ago.

**Comment by mkunz7** [comment]
I changed my ip address and setup my installation from latest. Will know the 29th if it worked.

**Comment by rusd11** [comment]
I recently set up my instance on Heroku, but did not have success. I have multiple upcoming checkins, so I can implement any recommended changes and test quickly.

The emails went to the _Enqueued_ list, while the checkins remained in the _Scheduled_ list. The emails did not send, but that is not my highest priority. The checkin was also unsuccessful as you can see they remained in the _Scheduled_ list.

The website UI and database all seem to be functioning correctly.

My current Heroku setup is as follows:
**web** bin/rails server -p $PORT -e $RAILS_ENV **"ON"**
**worker** bundle exec rake jobs:work **"ON"**

I currently have the _Free_ edition of Heroku, but noticed they say "_SLEEPS AFTER 30 MINS OF INACTIVITY_." Does this make a difference?

Also, I have the following add-ons:
Heroku Postgres :: Database	Hobby Dev	Free
Mailgun	Starter	Free
New Relic APM :: Newrelic	Wayne	Free
Redis To Go	Nano	Free

**Comment by flyize** [comment]
Yes, the sleeping matters.  You can use something like uptime robot to make Heroku stay awake.  However, you only get 1000 hours every month.  After that, Heroku puts them to sleep until the next month.

**Comment by wootwoot1234** [comment]
Isn't 1000 hours more than enough?  Should be over 40 days worth unless I'm not understanding that right?

I've been using [this service](http://kaffeine.herokuapp.com/) to keep my apps awake.  Seems to work well.

**Comment by flyize** [comment]
There are two dynos for this: web and worker.  So you're sucking up 48 hours every day.

**Comment by wootwoot1234** [comment]
I see.  What about using a different free hosting service like [openshift](https://www.openshift.com/)?  I've been using them for other projects for years and I don't think they don't have any limits for small projects like this.

**Comment by wootwoot1234** [comment]
Do both ```web``` and ```worker``` dynos need to be on?  I used the deploy button and the ```worker``` wasn't on by default.

**Comment by flyize** [comment]
From my understanding, they both need to be on.

**Comment by rusd11** [comment]
Check in this evening was successful on the @flyize instance, received A32. Looks like the difference between the two is the **web** and **worker* config. Yes, both dynos need to be running for everything to work.

**Comment by mkunz7** [comment]
My checkin worked today. Not using heroku.

**Comment by snekse** [comment]
I'm using the free tier on Heroku and the "Deploy to Heroku" button didn't fully work out-of-the-box.  The Worker dyno was created, but not started. The UI control on the dashboard wouldn't let me turn it on.  I had to run `heroku ps:scale worker+1 --app <MY-APP-NAME>` via the Heroku CLI to add a worker dyno slot.

**Comment by aortbals** [comment]
TLDR; the worker must be running at all times for the checkin process to work.

Just to chime in here: in order for this to work, **the worker has to be running at all times**. Checkins are scheduled and executed in the background on the worker, so if it isn't running, then nothing will happen. On heroku, from my understanding, you're going to have issues using the free tier because it will default to only running the web instance, and it also will cycle of dynos that are idle.

**Comment by flyize** [comment]
@aortbals Any ideas if the web dyno also has to be running for checkins to work?

And the half*** way around keeping the dynos up is to use an uptime monitor.  Pinging it every 5 minutes will prevent them from sleeping.

**Comment by aortbals** [comment]
> @aortbals Any ideas if the web dyno also has to be running for checkins to work?

No, it doesn't technically need to be running for a checkin to work.

> And the half*** way around keeping the dynos up is to use an uptime monitor. Pinging it every 5 minutes will prevent them from sleeping.

Ya, you can do that. If you want to really pay attention to your free dyno hours, then you can spin up both the web and worker to add a checkin, spin down the web worker, and then spin up the worker sometime before the checkin time. I personally think that's a pain and unreliable, so I would recommend using a low-resource VPS like a $5 digital ocean, but a lot of people are not comfortable with setting up a VPS, so that's a trade off.

**Comment by flyize** [comment]
Oh I can setup a VPS with my eyes closed.  However, after spending hours trying to get this to run on a standalone server, I gave up and was forced to use the one-click Heroku method.

**Comment by wootwoot1234** [comment]
@flyize I use [Kaffeine](http://kaffeine.herokuapp.com/) to keep my heroku app alive and have it turn off between 11pm and 5am because there aren't many southwest flights at that time anyway.

**Comment by flyize** [comment]
Oh, Kaffeine seems nice.  I don't see the option of when to turn it back on though...

**Comment by wootwoot1234** [comment]
It turns it off for 6 hours so you just set the start time.

**Comment by flyize** [comment]
Hmmm, is there a way to keep the worker on, but turn down the web if no one was accessing it?  If so, it would be pretty easy to stay under 1000 hours.

**Comment by wootwoot1234** [comment]
Not that I know of.

**Comment by mkunz7** [comment]
@flyize @wootwoot1234 I made a guide for debian 7 x64

VMs are pretty cheap, I don't see a real reason to keep abusing heroku's free plan.

**Comment by flyize** [comment]
Technically its working out for them in my case.  When I run out of hours, I turn on Hobby until the next month, so they get to make a couple of bucks off me.  

I'll check out your guide though.

**Comment by aortbals** [comment]
Closing this, as it is generally an issue with making sure sidekiq is running at all times for checkins, and not a bug with the checkin process.
